histomicstk/segmentation/label/trace_object_boundaries.py

```diff
@@ -38,19 +39,84 @@ def trace_object_boundaries(im_label,
     Returns
     -------
     X : array_like
-        A 1D array of horizontal coordinates of contour seed pixels for
+        A set of 1D array of horizontal coordinates of contour seed pixels for
         tracing.
     Y : array_like
-        A 1D array of the vertical coordinates of seed pixels for tracing.
+        A set of 1D array of the vertical coordinates of seed pixels for
+        tracing.
 
     References
     ----------
-    .. J. Seo et al "Fast Contour-Tracing Algorithm Based on a Pixel-
-       Following Method for Image Sensors" in Sensors,vol.16,no.353,
-       doi:10.3390/s16030353, 2016.
-
+    .. [1] J. Seo et al "Fast Contour-Tracing Algorithm Based on a Pixel-
+    Following Method for Image Sensors" in Sensors,vol.16,no.353,
+    doi:10.3390/s16030353, 2016.
     """
 
-    return trace_boundaries_cython(
-        np.ascontiguousarray(im_label, dtype=np.int),
-        conn, trace_all, x_start, y_start, max_length)
+    if max_length is None:
+        max_length = float('inf')
+
+    X = []
+    Y = []
+
+    if trace_all:
+
+        rprops = regionprops(im_label)
+        numLabels = len(rprops)
+
+        x_start = -1
+        y_start = -1
+
+        for i in range(numLabels):
```

Comment:
Yeah, just wanted to leave this one because there are some places where the loop is performed as similar pattern. Maybe later we could make a cython function that handles this pattern.